crates/primitives/src/chain/spec.rs

```diff
@@ -708,10 +708,10 @@ impl ForkTimestamps {
 #[derive(Debug, Clone, Deserialize, Serialize)]
 #[serde(untagged)]
 pub enum AllGenesisFormats {
-    /// The geth genesis format
-    Geth(Genesis),
     /// The reth genesis format
     Reth(ChainSpec),
+    /// The geth genesis format
+    Geth(Genesis),
```

Comment:
Changing the order here works @Rjected

Reply:
From [serde docs](https://serde.rs/enum-representations.html#untagged):
> There is no explicit tag identifying which variant the data contains. Serde will try to match the data against each variant in order and the first one that deserializes successfully is the one returned.

I added a test to ensure `Genesis` is deserialized correctly for `AllGenesisFormats` as well.

Reply:
ptal @Rjected